ui/raidboss/data/05-shb/raid/e9s.js

```diff
@@ -369,27 +370,65 @@ export default {
       // to have this cleaned up before the second Second Art Of Darkness
       preRun: (data) => delete data.finalArtOfDarkness,
       infoText: (data, _, output) => output.text(),
-      outputStrings: { text: artOfDarknessOutputStrings.startRight },
-      run: (data) => data.artOfDarkness = [],
+      outputStrings: {
+        text: {
+          en: 'Start Right',
+          de: 'Starte Rechts',
+        },
+      },
+      run: (data) => {
+        data.artOfDarkness = [];
+        if (!data.artOfDarknessIdMap)
+          data.artOfDarknessExpected = 'left';
+      },
     },
     {
-      id: 'E9S The Second / Third Art Of Darkness Stack Charge',
-      netRegex: NetRegexes.headMarker({ id: '01B9', capture: false }),
-      preRun: (data) => data.artOfDarkness.push('stackWithPartner'),
-      durationSeconds: (data) => data.finalArtOfDarkness ? 16 : 9,
-      alertText: (data, _, output) => {
-        // Perform the callout now, regardless if it's The Second or Third Art Of Darkness
-        if (data.finalArtOfDarkness)
-          data.artOfDarkness.push(data.finalArtOfDarkness);
-        return data.artOfDarkness.map((key) => output[key]()).join(' -> ');
+      // The Art Of Darkness uses head markers with randomized offsets.  The first
+      // charge is always left or right, and we can solve the rest from there.
+      id: 'E9S The Second / Third Art Of Darkness Charge Solver',
```

Comment:
Question here: When I run this in the emulator, it works perfectly. I'm presuming because this trigger is before the Right / Left and Callout triggers. I guess I was expecting to have to have done something to this trigger to add the output to the data.artOfDarkness array within the run, since the triggers have opposing conditions and it didn't feel like they'd both trigger off the same headmarker in order.

Reply:
I"m not sure what you mean by "it didn't feel like they'd both trigger off the same headmarker in order".  It is definitely the case that triggers are ordered, and two triggers that match the same line will be processed in the same order they are in the file.  (This is also why order has to be carefully preserved when users override triggers.)  Does that answer your question?

Reply:
> Does that answer your question?

Yep! Thanks for the explanation!